### ISSUE TITLE
problem with the CLI on MacOS

### ISSUE THREAD

**Issue by finedays** [Microsoft/PowerBI-Cli#5]
Hi guys, 
I am getting this error: 
`env: node\r: No such file or directory`
Using Iterm2, using zsh. Does anyone has any suggestions how to fix it?
Thanks

**Comment by wbreza** [comment]
Can you provide more context?  Which command are you executing?

**Comment by finedays** [comment]
Hi,

I am trying to execute powerbi from the shell. [https://www.dropbox.com/s/zrgt2xxu7k7q9ux/Screenshot%202016-05-04%2020.50.23.png?dl=0](url)

**Comment by wbreza** [comment]
Did you install the module globally?  `npm install -g powerbi-cli` ??

**Comment by finedays** [comment]
yes, I did.. it is exactly the same reaction when I try to run the cli from the /bin directory..

**Comment by wbreza** [comment]
Thanks for submitting this.  We'll get a fix out soon.

**Comment by finedays** [comment]
Thanks, looking forward the solution.

**Comment by wbreza** [comment]
FYI - Appears to be an issue with line endings between Windows and OSX.
http://stackoverflow.com/questions/18492897/env-node-r-no-such-file-or-directory-with-cordova-cli

**Comment by finedays** [comment]
Hm, that might work out. Thanks! I will try it in an hour or so and will message again.

**Comment by finedays** [comment]
Actually, I tried it now and it is working! Thanks!

**Comment by wbreza** [comment]
I'm going to keep this open for now till we can put a permanent fix that doesn't require this workaround.

**Comment by wbreza** [comment]
Reopening do to issue executing sub commands.

**Comment by karolkulesza** [comment]
Same issue here (not able to execute any command, resulting in "{command}(0) does not exist" error, in case of OS X.

Does this mean that PowerBI-Cli does not work on OS X, currently?

**Comment by wbreza** [comment]
@karolkulesza - Yes - there is currently a bug that we are working through.

**Comment by homburgs** [comment]
Same bug here under Ubuntu Linux 14.04 LTS
with nodejs 4.5.0

**Comment by jbb352** [comment]
Any update on a release on this?

**Comment by stevenharman** [comment]
FWIW, the hack around this is to fix the line endings of all of the files in the `power-bi` Node modules.

```shell
$ brew install dos2unix
$ find /usr/local/lib/node_modules/powerbi-cli -name "*.js" | xargs -n 1 sudo dos2unix
```

![do-it-live](https://cloud.githubusercontent.com/assets/48658/19052705/d0e2927c-8985-11e6-9109-f1f72d33233b.gif)

**Comment by stevenharman** [comment]
A hack which does "work..."

```shell
pushd /usr/local/lib/node_modules/powerbi-cli/bin
for f in cli-*; do mv "$f" "`echo $f | sed s/cli-/powerbi-/`"; done
popd
powerbi config
```

**Comment by BretStateham** [comment]
Any update on this?

**Comment by ali-hamud** [comment]
This task is in our backlog. no ETA.

**Comment by lipp** [comment]
@ali-hamud  I really like and appreciate to see that Microsoft is doing node. 
But why is this not on the backlog? The installation is basically not working at all / broken on OSX (and maybe other distros).

**Comment by gregbonnette** [comment]
FYI. This is also an issue within the Windows bash shell aka Ubuntu 14.04.4 LTS. 

The previous dos2unix workaround did not work.

`~$ powerbi config
execvp(): No such file or directory

  powerbi-config(1) does not exist, try --help`

**Comment by zimmermanw84** [comment]
FYI also broken on Ubuntu 16.04 LTS

![selection_017](https://cloud.githubusercontent.com/assets/8593005/25685025/80e6e134-3019-11e7-838c-dec91b70ccf1.png)
![whatface](https://cloud.githubusercontent.com/assets/8593005/25685069/c65994b4-3019-11e7-9cac-ead67564bd44.jpeg)

**Comment by adadgio** [comment]
Heys guys, any update or fix on this issue ? 😞

**Comment by agger-magenta** [comment]
I can confirm that the workaround involving dos2unix does not work under Ubuntu 16.04 LTS.
